### PR TITLE
Upgrade dredd-transactions to 4.4.0

### PR DESCRIPTION
**Pull request by michalholasek**
#### :rocket: Why this change?
This PR introduces `dredd-transactions` with `fury-adapter-swagger` version [0.14](https://github.com/apiaryio/dredd-transactions/pull/108).

#### :memo: Related issues and Pull Requests

- Fixes https://github.com/apiaryio/dredd/issues/892

#### :white_check_mark: What didn't I forget?

- [x] To write docs - N/A
- [ ] To write tests
- [x] To put [Conventional Changelog](https://dredd.readthedocs.io/en/latest/contributing/#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

**Comment by honzajavorek** [comment]
All CIs failed. There's something broken with the behavior of Dredd with Swagger.

**Comment by michalholasek** [comment]
@honzajavorek Yep, I am looking into it.

File: test/integration/dredd-test.coffee [hunk 1]
```diff
@@ -562,17 +562,26 @@ describe 'Dredd class Integration', ->
...
+      assert.equal(matches[2][1], 'skip')
+    )
+    it('the transaction #4 is skipped', ->
+      assert.equal(matches[3][1], 'skip')
+    )
+    it('the transaction #5 is not skipped (status 200)', ->
+      assert.notEqual(matches[4][1], 'skip')
+    )
+    it('the transaction #6 is not skipped (status 200)', ->
+      assert.notEqual(matches[5][1], 'skip')
```
Comment (honzajavorek) [inline test/integration/dredd-test.coffee]: Good. But given the changes, I would turn this test into sth like

- when Swagger document has multiple responses
  - the responses with status code 200 are not skipped
  - the responses with status code 500 are skipped

I'd keep `recognizes all 6 transactions`, but the rest would iterate over expected values and inspect the matches more. I think this way we rely too much on what is in the fixture and what the structure is and the outcome of the test isn't very clear anymore.

File: test/integration/dredd-test.coffee [hunk 2]
```diff
@@ -592,18 +601,27 @@ describe 'Dredd class Integration', ->
...
     )
+    it('the transaction #4 is not skipped (status 200)', ->
+      assert.notEqual(matches[3][1], 'skip')
+    )
+    it('the transaction #5 is not skipped (status 200)', ->
+      assert.notEqual(matches[4][1], 'skip')
+    )
+    it('the transaction #6 is not skipped (unskiped in hooks)', ->
+      assert.notEqual(matches[5][1], 'skip')
+    )
```
Comment (honzajavorek) [inline test/integration/dredd-test.coffee]: Similar to above. Make an array of expectations about the individual matches and iterate.

```coffee
[
  {statusCode: 200, skipped: false, reason: 'unskipped by default'}
  {statusCode: 500, skipped: false, reason: 'unskipped in hooks'}
  ...
].forEach((expectation, i) ->
  context("the transaction ##{i}", ->
    it("has status code #{expectation.statusCode}", ->
      assert...
    )
    it("is #{if expectation.skipped then '' else 'not '}skipped (#{expectation.reason})", ->
      assert...
    )
  )
)
```

(Just out of my head, can contain mistakes)

**Comment by honzajavorek** [comment]
This needs to wait for https://github.com/apiaryio/dredd-transactions/pull/118 and then it should be revisited.

**Comment by honzajavorek** [comment]
Incorporated into https://github.com/apiaryio/dredd/pull/906.

**Pull request closed without merging**